workflows/scnanoseq.nf

```diff
@@ -365,267 +369,158 @@ workflow SCNANOSEQ {
+            )
+            ch_multiqc_finalqc_files = ch_multiqc_finalqc_files.mix(
+                PROCESS_LONGREAD_SCRNA_GENOME.out.dedup_idxstats.collect{it[1]}.ifEmpty([])
+            )
+        }
 
-        //
-        // MODULE: Samtools merge
-        //
-        ch_bams_to_merge = ch_dedup_bam
-                                .map{
-                                    meta, bam ->
-                                        bam_basename = bam.toString().split('/')[-1]
-                                        split_bam_basename = bam_basename.split(/\./)
-                                        meta = [ 'id': split_bam_basename[0] ]
-                                    [ meta, bam ]
-                                }
-                                .groupTuple()
-
-        SAMTOOLS_MERGE ( ch_bams_to_merge, fasta, fai)
-
-        ch_dedup_merged_bam = SAMTOOLS_MERGE.out.bam
-
-        // SUBWORKFLOW: BAM_SORT_STATS_SAMTOOLS
-        // The subworkflow is called in both the minimap2 bams and filtered (mapped only) version
-        BAM_SORT_STATS_SAMTOOLS_DEDUP ( ch_dedup_merged_bam,
-                                        fasta )
-
-        ch_dedup_sorted_bam = BAM_SORT_STATS_SAMTOOLS_DEDUP.out.bam
-        ch_dedup_sorted_bai = BAM_SORT_STATS_SAMTOOLS_DEDUP.out.bai
-
-        // these stats go for multiqc
-        ch_dedup_sorted_stats = BAM_SORT_STATS_SAMTOOLS_DEDUP.out.stats
-        ch_dedup_sorted_flagstat = BAM_SORT_STATS_SAMTOOLS_DEDUP.out.flagstat
-        ch_dedup_sorted_idxstats = BAM_SORT_STATS_SAMTOOLS_DEDUP.out.idxstats
-        ch_versions = ch_versions.mix(BAM_SORT_STATS_SAMTOOLS_DEDUP.out.versions)
+        ch_multiqc_finalqc_files = ch_multiqc_finalqc_files.mix(
+            ch_read_counts.collect().ifEmpty([])
+        )
+        ch_multiqc_finalqc_files = ch_multiqc_finalqc_files.mix(
+            PROCESS_LONGREAD_SCRNA_GENOME.out.gene_qc_stats.collect().ifEmpty([])
+        )
+        ch_multiqc_finalqc_files = ch_multiqc_finalqc_files.mix(
+            PROCESS_LONGREAD_SCRNA_GENOME.out.transcript_qc_stats.collect().ifEmpty([])
+        )
     }
 
-    //
-    // MODULE: Isoquant
-    //
-    ISOQUANT ( ch_dedup_sorted_bam.join(ch_dedup_sorted_bai, by: [0]), gtf, fasta, fai, 'tag:CB')
-    ch_gene_count_mtx = ISOQUANT.out.gene_count_mtx
-    ch_transcript_count_mtx = ISOQUANT.out.transcript_count_mtx
-    ch_versions = ch_versions.mix(ISOQUANT.out.versions)
+    // oarfish expects deduplicated reads
+    if (transcript_quants) {
+        PROCESS_LONGREAD_SCRNA_TRANSCRIPT (
+            transcript_fasta,
+            transcript_fai,
+            gtf,
+            ch_extracted_fastq,
+            ch_rseqc_bed,
+            ch_corrected_bc_info,
+            transcript_quants,
+            params.skip_save_minimap2_index,
+            params.skip_qc,
+            true,
+            params.skip_bam_nanocomp,
+            params.skip_seurat,
+            false,
+            false
```

Comment:
Disabling splitting the BAM in the transcriptomic data is currently creating a bottleneck at this step in regards to run time (BAM deduplication). This was originally solved in the genome mapping by splitting the BAM by chromosome, but due to the nature of the transcriptomic mappings, this cannot be enabled at this step as is.

@atrull314 and I are currently evaluating different approaches to implementing this optimization (or an alternative method), and once we have updated this portion of the workflow, the PR can move forward.